docs/design/keyboard_handler.md

```diff
@@ -0,0 +1,87 @@
+      break;
+      default:
+       std::cout << "callback with key code = " << static_cast<int32_t>(key_code) << std::endl;
+      break;
+    }
+  }
+```
+
+## Consideration of using C++ versus Python for cross-platform implementation
+At the very early design discussions was proposed to use Python as cross-platform 
+implementation for keyboard handling. From the first glance it looks attractive to use Python 
+since it's relatively new language with reach built-in utility functions available on multiple 
+platforms and OSes.
+However after some research and consideration we come to the conclusion that using Python is 
+not better than C or C++ for this case. And here is the rational for that:
+1. It turns out that there are no built-in and cross-platform utility functions for handling input 
+   from keyboard in Python. All what you can find is a third party libraries with similar 
+   compile time division for Windows and Unix platforms. Most of the libraries not mature 
+   enough and have low quality.
+2. Most of the ROS2 code written in C++ and it will be much easy to support and use solution
+   written on the same language.
+3. Some of the POSIX compatible OSes doesn't have Python interpreter. For instance [QNX for safety](https://blackberry.qnx.com/en/software-solutions/embedded-software/qnx-os-for-safety) 
+
+## Specific of handling input from keyboard on different platforms
+Processing of the keyboard handling is differ for different operating systems.
+There are two major different approaches:
+1. Unix/MacOS aka POSIX "compatible" OSes
+2. Windows family OSes
+
+Posix compatible OSes approach based on readout from standard input linked to the terminal.
+Pressed key combination represented as sequence of 8 bit values. Number of the 8 bit values in sequence
+could vary from pressed key combination and usually in range of 1-8 chars. Note: There are no
+'\0' value representing null terminator in providing sequence of bytes. i.e. those byte
+sequence could be treated as "C" strings if add null terminator at the end.
+
+In Windows family OSes available mechanism for keyboard handling derived from DOS which is based on
+polling "`int kbhit()`" system function call until it will return non zero value indicating that
+some key was pressed. After that need to read integer value(s) from the standard input to
+determine which key was pressed. Pressed key combinations could be represented as one or two 
+integer values. Simple alphanumerical keys represented as single integer. Control keys and specific
+function keys represented as two integer values. Note: for some key combinations first integer 
+value could contain '\0' value representing null terminator in regular strings. 
+i.e it is not possible to treat such sequence of bytes as regular strings.
+
+## Considerations of how to handle cases when client code subscribed to the key press event got destructed before keyboard handler.
+TBD.
```

Comment:
By returning some handle to the subscription that the client has to own, you can have it unregister itself in the destructor.